skglm/penalties/generalized_linear_cv.py

```diff
@@ -0,0 +1,147 @@
+
+    def fit(self, X, y):
+        """Fit the model using cross-validation."""
+        if not hasattr(self.penalty, "alpha"):
+            raise ValueError("'penalty' must expose an 'alpha' hyper-parameter.")
+        X, y = np.asarray(X), np.asarray(y)
+        n, p = X.shape
+        rng = np.random.RandomState(self.random_state)
+
+        if self.alphas is not None:
+            alphas = np.sort(self.alphas)[::-1]
+        else:
+            alpha_max = np.max(np.abs(X.T @ y)) / n
+            alphas = np.logspace(
+                np.log10(alpha_max),
+                np.log10(alpha_max * self.eps),
+                self.n_alphas
+            )[::-1]
+        has_l1_ratio = hasattr(self.penalty, "l1_ratio")
+        l1_ratios = [1.] if not has_l1_ratio else np.atleast_1d(
+            self.l1_ratio if self.l1_ratio is not None else [1.])
+
+        folds = list(_kfold_split(n, self.cv, rng))
+        n_folds = len(folds)
+
+        mse_path = np.empty((len(l1_ratios), len(alphas), n_folds))
+        best_loss = np.inf
+
+        def _solve_fold(k, train, test, alpha, l1, w_start):
+            pen_kwargs = {k: v for k, v in self.penalty.__dict__.items()
+                          if k not in ("alpha", "l1_ratio")}
+            if has_l1_ratio:
+                pen_kwargs['l1_ratio'] = l1
+            pen = type(self.penalty)(alpha=alpha, **pen_kwargs)
+
+            kw = dict(X=X[train], y=y[train], datafit=self.datafit, penalty=pen)
+            if 'w' in self.solver.solve.__code__.co_varnames:
+                kw['w'] = w_start
+            w = self.solver.solve(**kw)
+            w = w[0] if isinstance(w, tuple) else w
+
+            coef, intercept = (w[:p], w[p]) if w.size == p + 1 else (w, 0.0)
+
+            y_pred = X[test] @ coef + intercept
+            return w, self._score(y[test], y_pred)
+
+        for idx_ratio, l1_ratio in enumerate(l1_ratios):
+            warm_start = [None] * n_folds
+
+            for idx_alpha, alpha in enumerate(alphas):
+                fold_results = Parallel(self.n_jobs)(
+                    delayed(_solve_fold)(k, tr, te, alpha, l1_ratio, warm_start[k])
+                    for k, (tr, te) in enumerate(folds)
+                )
+
+                for k, (w_fold, loss_fold) in enumerate(fold_results):
+                    warm_start[k] = w_fold
+                    mse_path[idx_ratio, idx_alpha, k] = loss_fold
+
+                mean_loss = np.mean(mse_path[idx_ratio, idx_alpha])
+                if mean_loss < best_loss:
+                    best_loss = mean_loss
+                    self.alpha_ = float(alpha)
+                    self.l1_ratio_ = float(l1_ratio) if l1_ratio is not None else None
+
+        # Refit on full dataset
+        self.penalty.alpha = self.alpha_
+        if hasattr(self.penalty, "l1_ratio"):
+            self.penalty.l1_ratio = self.l1_ratio_
+        super().fit(X, y)
+        self.alphas_ = alphas
+        self.mse_path_ = mse_path
+        return self
```

Comment:
Look at how LassoCV works: after fitting, you can directly call it to predict, because it stores the best Lasso. you need to do the same, storing the best model coefficient, the best  `best_alpha_ ` and recreating a GeneralizedLienarModel at the end, that fits on the whole dataset X:

look at how it's done here : 
https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/linear_model/_coordinate_descent.py#L1882